src/lib.rs

```diff
@@ -706,6 +726,94 @@ where
         }
     }
 
+    /// Removes multiple of a value from the bag. If `quantity` is greater than the number of
+    /// occurences, zero occurances will remain.
+    ///
+    /// The number of occurrences of the value currently in the bag is returned.
+    ///
+    /// The value may be any borrowed form of the bag's value type, but
+    /// [`Hash`] and [`Eq`] on the borrowed form *must* match those for
+    /// the value type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hashbag::HashBag;
+    ///
+    /// let mut bag = HashBag::new();
+    ///
+    /// bag.insert_many('x', 10);
+    /// assert_eq!(bag.contains(&'x'), 10);
+    /// assert_eq!(bag.remove_many(&'x', 3), 7);
+    /// assert_eq!(bag.contains(&'x'), 7);
+    /// assert_eq!(bag.remove_many(&'x', 10), 0);
+    /// ```
+    #[inline]
+    pub fn remove_many<Q: ?Sized>(&mut self, value: &Q, quantity: usize) -> usize
+    where
+        T: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        match self.items.get_mut(value) {
+            None => 0,
+            Some(n) if *n <= quantity => {
+                self.count -= 1;
+                self.items.remove(value);
+                0
+            }
+            Some(n) => {
+                *n -= quantity;
+                *n
+            }
+        }
+    }
+
+    /// Removes multiple of a value from the bag. If `quantity` is greater than the number of
+    /// occurences, then the value is unchanged
```

Comment:
I think we should just leave this method out. It feels like a corner case where we don't have a great sense of this even being the right kind of signature someone with that use might want.

Reply:
Sure, I can remove it. The idea of "remove many" was somewhat vague, so I figured over-defining was better than picking a single definition.

Reply:
I agree with that, I'd just prefer to only land the method we _know_ we need in this instance, and do the over-defining in the name instead.